pkgs/tools/misc/rtx/default.nix

```
@@ -0,0 +1,71 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, stdenvNoCC
+, perl
+, coreutils
+, bash
+, direnv
+, Security
+}:
+
+let
+  owner = "jdxcode";
+  pname = "rtx";
+  version = "1.29.7";
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-w61ik1Tne7cO1Aez4aem3/UQfWjoCTnz+vR1YaaHDhY=";
+  };
+
+  cargoSha256 = "sha256-PGq/LSM8bsNWRVDLLhaiEgaXIN+957HnF0SUx6M7Wus=";
+
+  # perl because: https://docs.rs/openssl/0.10.45/openssl/#vendored
+  nativeBuildInputs = [ perl installShellFiles ];
+  buildInputs = lib.optionals stdenvNoCC.isDarwin [ Security ];
+
+  # see for reference: https://github.com/jdxcode/rtx/blob/main/default.nix
+  # upstream start
+  prePatch = ''
+    substituteInPlace ./test/data/plugins/**/bin/* \
+      --replace '#!/usr/bin/env bash' '#!${bash}/bin/bash'
+    substituteInPlace ./src/fake_asdf.rs ./src/cli/reshim.rs \
+      --replace '#!/bin/sh' '#!${bash}/bin/sh'
+    substituteInPlace ./src/env_diff.rs \
+      --replace '"bash"' '"${bash}/bin/bash"'
+    substituteInPlace ./src/cli/direnv/exec.rs \
+      --replace '"env"' '"${coreutils}/bin/env"' \
+      --replace 'cmd!("direnv"' 'cmd!("${direnv}/bin/direnv"'
+  '';
+
+  # Skip the test_plugin_list_urls as it uses the .git folder, which
+  # is excluded by default from Nix.
+  checkPhase = ''
+    RUST_BACKTRACE=full cargo test --all-features -- \
+      --skip cli::plugins::ls::tests::test_plugin_list_urls
+  '';
```

nit: Ensure tests run in release mode

> To make sure that we don’t compile the sources twice and to actually test the artifacts that will be used at runtime, the tests will be ran in the `release` mode by default.

https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo

```suggestion
  checkPhase = ''
    RUST_BACKTRACE=full cargo test --release --all-features -- \
      --skip cli::plugins::ls::tests::test_plugin_list_urls
  '';
```

Using the modules arguments would take care of it among other things (like parallelism):

```suggestion
  cargoTestFlags = [ "--all-features" ];
  checkFlags = ["--skip=cli::plugins::ls::tests::test_plugin_list_urls"];
```

You can review the check hook here if you'd like: [`master`/pkgs/build-support/rust/hooks/cargo-check-hook.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/hooks/cargo-check-hook.sh)

---

ahhh, yes, wanted that and learned a thing but...

Some tests started failing on me, and turns out that the release mode was not the cause (as I initially thought) but parallelism. Needed to disable it since some test are accessing the same folder (as far as I understand). So I needed to set this:
```nix
  # some tests access the same folders, don't test in parallel to avoid race conditions
  dontUseCargoParallelTests = true;
```

The commit is amended, please take a look.